### PR TITLE
Fix deprecations in SoftCosineSimilarity

### LINKED ISSUE #2938: Fix deprecations in SoftCosineSimilarity
When running our CI test suite, I see an array of deprecation warnings:

https://dev.azure.com/rare-technologies/gensim-ci/_build/results?buildId=287&view=logs&jobId=f9575ddc-dec8-54e6-9d26-abb8bdd9bed7&j=f9575ddc-dec8-54e6-9d26-abb8bdd9bed7&t=180156a9-2bf9-537d-c84a-ef9e808c0367

Some are from gensim, some from scipy:
Screen Shot 2020-09-09 at 09 16 00

@Witiko could you please have a look? Is it something your existing PR already addresses?

If not, can you please fix those? Thanks.

### PR DESCRIPTION
**Pull request by Witiko**
This PR removes deprecated Soft Cosine Measure parameters, functions, and tests, and closes #2938.

Here is a detailed list of the deprecations:
- Parameter `positive_definite` of `SparseTermSimilarityMatrix` has been renamed to `dominant`. Test `test_positive_definite` has been removed.
- Parameter `similarity_matrix` of `SoftCosineSimilarity` no longer accepts unencapsulated sparse matrices, only `SparseTermSimilarityMatrix` objects.
- Parameter `normalized` of `SparseTermSimilarityMatrix.inner_product` no longer accepts booleans, only two-tuples.
- Function `matutils.softcossim` has been superseded by method `SparseTermSimilarityMatrix.inner_product`. Tests in   `TestSoftCosineSimilarity` have been removed.

### CODE REVIEW

**Comment by piskvorky** [comment]
Great! Thanks.

I know it's not related to SoftCossim, but would you mind checking the other deprecations too?
They seem mostly syntactic: recent numpy's `tostring()` => `tobytes`, `smart_open.smart_open()` => `smart_open.open()` etc. It'll help us make the 4.0.0 release cleaner.

File: gensim/test/test_similarity_metrics.py [hunk 1]
```diff
@@ -240,36 +240,6 @@ def test_distributions(self):
         self.assertAlmostEqual(expected, result)
 
 
-class TestSoftCosineSimilarity(unittest.TestCase):
```
Comment (piskvorky) [inline gensim/test/test_similarity_metrics.py:243]: How is `softcossim`'s replacement, `inner_product` itself, tested? Did we use to double-test it?

Comment (Witiko) [reply gensim/test/test_similarity_metrics.py:243]: There are 63 unit tests in [`test_similarities.py`](https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/test/test_similarities.py#L1039).

**Comment by Witiko** [comment]
@piskvorky Perhaps I should create a separate PR for these other deprecation warnings that are coming from outside of Gensim's code and are unrelated to the soft cosine measure?

**Comment by piskvorky** [comment]
No difference. Fixing them will be awesome either way.

I'm thinking we should up our minimum numpy to 1.9.0, to avoid having to work around the different names with code branching. 1.9.0 is sufficiently old that we don't have to worry about compatibility (I think).

**Comment by piskvorky** [comment]
Never mind – I see we already require numpy >= 1.11, so `tobytes()` is guaranteed to exist.

**Comment by Witiko** [comment]
I have fixed most of the deprecation warnings in e724bd0.

**Comment by mpenkov** [comment]
Merged. Thank you @Witiko for your contribution and your patience!

**Pull request merged**